src/snowflake/snowpark/_internal/analyzer/analyzer.py

```diff
@@ -1041,7 +1047,9 @@ def do_resolve_with_resolved_children(
                 )
                 if logical_plan.condition
                 else None,
-                resolved_children.get(logical_plan.source_data, None),
```

Comment:
This is not introduced in this PR, but I'm curious why we need to do `resolved_children.get(logical_plan.source_data, None)`. 

From the references to `TableUpdate`/`TableDelete`/`TableMerge`, `source` or `source_data` will already be resolved. In addition, `logical_plan.source_data` is guaranteed to be in `resolved_children`. https://github.com/snowflakedb/snowpark-python/blob/697294d8ee89396deab4309dea384b1eadb75d6b/src/snowflake/snowpark/_internal/analyzer/table_merge_expression.py#L54 

I feel like just `logical_plan.source_data` would suffice.

Reply:
cc: @sfc-gh-jdu